src/main/java/com/appland/appmap/record/RecordingSessionFileStream.java

```diff
@@ -28,7 +33,20 @@
   public RecordingSessionFileStream(String fileName, Metadata metadata) {
     this.metadata = metadata;
     if (fileName != null) {
-      this.fileName = String.format("%s.appmap.json", fileName);
+      if (fileName.length() + DEFAULT_FILENAME.length() >=  FILENAME_MAX_LENGTH) {
```

Comment:
This logic is valid, but `DEFAULT_FILENAME` is meant as the default name of the file to be emitted if none is given otherwise. It's subject to change, and shouldn't affect this logic. Maybe we need something more like a static `FILE_EXTENSION` variable?

Reply:
roger that!